velox/exec/DistinctAggregations.cpp

```diff
@@ -126,12 +131,62 @@ class TypedDistinctAggregations : public DistinctAggregations {
   }
 
  private:
+  bool isSingleInputAggregate() const {
+    return aggregates_[0]->inputs.size() == 1;
+  }
+  void decodeInput(const RowVectorPtr& input, const SelectivityVector& rows) {
+    inputForAccumulator_ = makeInputForAccumulator(input);
+    decodedInput_.decode(*inputForAccumulator_, rows);
+  }
+
+  static TypePtr makeInputTypeForAccumulator(
+      const RowTypePtr& rowType,
+      const std::vector<column_index_t>& inputs) {
+    if (inputs.size() == 1) {
+      return rowType->childAt(inputs[0]);
+    }
+
+    // Otherwise, synthesize a ROW(distinct_channels[0..N])
+    std::vector<TypePtr> types;
+    std::vector<std::string> names;
+    for (column_index_t channelIndex : inputs) {
+      names.emplace_back(rowType->nameOf(channelIndex));
+      types.emplace_back(rowType->childAt(channelIndex));
+    }
+    return ROW(std::move(names), std::move(types));
+  }
+
+  VectorPtr makeInputForAccumulator(const RowVectorPtr& input) const {
+    if (isSingleInputAggregate()) {
+      return input->childAt(inputs_[0]);
+    }
+
+    std::vector<VectorPtr> newChildren(inputs_.size());
+    for (int i = 0; i < inputs_.size(); ++i) {
+      newChildren[i] = input->childAt(inputs_[i]);
+    }
+    return std::make_shared<RowVector>(
+        pool_, inputType_, nullptr, input->size(), newChildren);
+  }
+
+  std::vector<VectorPtr> makeInputForAggregation(
+      std::shared_ptr<BaseVector> input) const {
+    std::vector<VectorPtr> args;
+    if (isSingleInputAggregate()) {
+      args = {std::move(input)};
+      return args;
+    }
+    return input->template asUnchecked<RowVector>()->children();
+  }
+
+  memory::MemoryPool* const pool_;
   const std::vector<AggregateInfo*> aggregates_;
-  const column_index_t input_;
+  const std::vector<column_index_t> inputs_;
   const TypePtr inputType_;
-  memory::MemoryPool* const pool_;
 
   DecodedVector decodedInput_;
+  VectorPtr inputForAccumulator_;
+  std::vector<VectorPtr> inputForAggregation_;
```

Comment:
Does this need to be a member variable? Looks like this can be a local variable, no?